pilot/pkg/model/service.go

```diff
@@ -1272,6 +1272,22 @@ func (s *Service) GetExtraAddressesForProxy(node *Proxy) []string {
 	return nil
 }
 
+// GetAllAddressesForProxy returns a k8s service's extra addresses to the cluster where the node resides.
+// Especially for dual stack k8s service to get other IP family addresses.
+func (s *Service) GetAllAddressesForProxy(node *Proxy) []string {
+	if features.EnableDualStack && node.Metadata != nil {
+		if node.Metadata.ClusterID != "" {
```

Comment:
would it make sense to fallback to getAddressForProxy(node) if node.Metadata.ClusterID == "" ?